docs/building.md

```diff
@@ -0,0 +1,167 @@
+  **must be specified**.
+
+#### Building and flashing firmware to Fobnail Token
+
+For hardware setup instructions see [Flashing preparation](flashing_preparation.md).
+
+Building and flashing is performed by executing (from `fobnail` directory):
+
+```shell
+env FOBNAIL_PO_ROOT=root_ca.crt FOBNAIL_EK_ROOT_DIR=tpm_ek_roots \
+    ./build.sh -t nrf --run
+```
+
+A console with Fobnail Token output will be displayed. It is not required for
+normal operation but can be useful for debugging. It can be closed with Ctrl-C
+at any point. After that, Token can be used without nRF52840-DK - just plug it
+wherever it's needed.
+
+#### Building and running firmware locally
+
+Make sure you have the network set up properly. See [Networking setup](networking_setup.md)
+for instructions. For firmware to work properly you need the `fobnail0` network
+interface configured.
+
+To build and run firmware execute the following commands (from `fobnail`
+directory).
+
+```shell
+env FOBNAIL_LOG=info FOBNAIL_PO_ROOT=root_ca.crt \
+    FOBNAIL_EK_ROOT_DIR=tpm_ek_roots ./build.sh -t pc --run
+```
+
+`FOBNAIL_LOG` environment variable sets log level. The possible log levels are:
+`error`, `warning`, `info`, `debug`, `trace`. This variable is valid only for PC
+target.
+
+Another variable used only on PC is `FOBNAIL_DEVICE_ID` - on hardware we used
+FICR registers to create a device ID, this gives a way of configuring it for
+emulation. This variable is optional, without it an ID of 0 is used.
+
+`build.sh` automatically starts Token emulation. It runs until it's terminated
+with Ctrl-C.
+
+## PC applications
+
+Platform Owner and Attester have similar set of prerequisites, listed below.
+Depending on use case, they may or may not be run on the same PC, so these steps
+may have to be repeated on different computers.
+
+```shell
+sudo apt -y install git make gcc autoconf automake pkg-config libtool libssl-dev
+```
+
+Both applications use [libcoap with v3 API](https://github.com/obgm/libcoap/tree/release-4.3.0)
+which isn't provided by main Linux distributions yet, so it has to be built from
+sources and installed:
+
+```shell
+git clone https://github.com/obgm/libcoap.git --recurse-submodules
+cd libcoap
+git checkout release-4.3.0
+./autogen.sh && \
+./configure --exec-prefix=/usr --disable-tests --disable-documentation \
+    --disable-manpages --enable-dtls --with-tinydtls --enable-fast-install && \
+make && sudo make install
+```
+
+#### Platform Owner
+
+With prerequisites installed, building Platform Owner application is simple -
+just clone and build:
+
+```shell
+git clone https://github.com/fobnail/fobnail-platform-owner --recurse-submodules
+cd fobnail-platform-owner
+make
+```
+
+Produced binary is located in `bin/fobnail-platform-owner`, from where it can be
+moved to `PATH` or just started from there.
+
+#### Attester
+
+Attester requires additional packages to talk with TPM and download TPM's
+certificate chain from Internet:
```

Comment:
Excellent. What does it mean, and how can I fix it? Here, there should be a link to the TPM simulator or an explanation of the user's options.

This is indirectly explained in the section `Environment variables common for both targets,` but we should simplify things for the user and explain what is next. TBH, both sections, Platform Owner and Attester, give no value, ok I compiled something, but what is it, and how to use it after all those steps I made?

Reply:
This doesn't belong to `Cloning and building` instructions. I will add summary with links to next steps, I just have to think how to present it. From this point instructions between `pc`/`nrf` and real/emulated TPM will be slightly different.

Reply:
Added pointers to next steps in https://github.com/fobnail/docs/pull/47/commits/a35de16316f492d7dfd93007715a3cf335a40ab3